include/psa/crypto_platform.h

```diff
@@ -46,57 +46,46 @@
 /* PSA requires several types which C99 provides in stdint.h. */
 #include <stdint.h>
 
+#include <psa/crypto.h>
+
 /* Integral type representing a key handle. */
 typedef uint16_t psa_key_handle_t;
 
-/* This implementation distinguishes *application key identifiers*, which
- * are the key identifiers specified by the application, from
- * *key file identifiers*, which are the key identifiers that the library
- * sees internally. The two types can be different if there is a remote
- * call layer between the application and the library which supports
- * multiple client applications that do not have access to each others'
- * keys. The point of having different types is that the key file
- * identifier may encode not only the key identifier specified by the
- * application, but also the the identity of the application.
- *
- * Note that this is an internal concept of the library and the remote
- * call layer. The application itself never sees anything other than
- * #psa_app_key_id_t with its standard definition.
- */
-
-/* The application key identifier is always what the application sees as
- * #psa_key_id_t. */
-typedef uint32_t psa_app_key_id_t;
-
-#if defined(MBEDTLS_PSA_CRYPTO_KEY_FILE_ID_ENCODES_OWNER)
+#if defined(MBEDTLS_PSA_CRYPTO_KEY_EXTENDED_ID)
 
 #if defined(PSA_CRYPTO_SECURE)
 /* Building for the PSA Crypto service on a PSA platform. */
 /* A key owner is a PSA partition identifier. */
 typedef int32_t psa_key_owner_id_t;
+#else
+/* Default */
+typedef uint32_t psa_key_owner_id_t;
 #endif
 
 typedef struct
 {
-    uint32_t key_id;
-    psa_key_owner_id_t owner;
-} psa_key_file_id_t;
-#define PSA_KEY_FILE_GET_KEY_ID( file_id ) ( ( file_id ).key_id )
+    psa_key_owner_id_t owner_id;
+    psa_key_id_t owner_key_id;
+} psa_key_extended_id_t;
+
+#define PSA_KEY_ID_T psa_key_extended_id_t
+#define PSA_KEY_ID_INIT( owner_id, owner_key_id ) { \
+            (psa_key_owner_id_t)( owner_id ), \
+            (psa_key_id_t)( owner_key_id ) }
+#define PSA_KEY_ID_GET_ID( key_id ) ( ( key_id ).owner_key_id )
+#define PSA_KEY_ID_GET_OWNER_ID( key_id ) ( ( key_id ).owner_id )
 
-/* Since crypto.h is used as part of the PSA Cryptography API specification,
- * it must use standard types for things like the argument of psa_open_key().
- * If it wasn't for that constraint, psa_open_key() would take a
- * `psa_key_file_id_t` argument. As a workaround, make `psa_key_id_t` an
- * alias for `psa_key_file_id_t` when building for a multi-client service. */
-typedef psa_key_file_id_t psa_key_id_t;
-#define PSA_KEY_ID_INIT {0, 0}
+#else /* !MBEDTLS_PSA_CRYPTO_KEY_EXTENDED_ID */
 
-#else /* !MBEDTLS_PSA_CRYPTO_KEY_FILE_ID_ENCODES_OWNER */
+#define PSA_KEY_ID_T psa_key_id_t
```

Comment:
I did it like that for two reasons:
1) That way the signature of the PSA APIs in crypto.h for openless APIS (will) look close to what is defined in the PSA specification. For example,
psa_status_t psa_sign_hash(PSA_KEY_ID_T key, ... ) instead of psa_status_t psa_sign_hash(psa_key_id_t key, ...) in the spec.

2) I didn't find a good name for a type that would be equal to `psa_key_id_t` if  MBEDTLS_PSA_CRYPTO_KEY_FILE_ID_ENCODES_OWNER (or equivalent) is not defined and `psa_key_extended_id_t` (or equivalent) otherwise.

Thinking about this again, I think I'd prefer just having `psa_key_id_t` not defined as in the PSA spec (in case where key identifiers encode an owner identifier) and all the PSA openless API signatures defined as in the PSA spec, rather than the situation induced by the current status of this PR where `psa_key_id_t` is always as defined in the PSA spec but all the PSA openless API signatures are(will) not (again, in case where key identifiers encode an owner identifier). I'd prefer the first solution because there would just one deviation from the spec in the interface with expose. We already discussed about that and I think you were in favor of the second solution if I remember/understood correctly. Is it the case? Why?

Reply:
1. With a macro, there's `PSA_KEY_ID_T` instead of `psa_key_id_t`. With a type, there's `psa_key_id_with_owner_t` instead of `psa_key_id_t`. Same difference.
2. With a type, you need to find a good name for the type. With a macro, you need to find a good name for the macro. Same difference.

I prefer to keep `psa_key_id_t` defined as in the PSA specification because I think it makes the code misleading otherwise. The reason `crypto.h` currently uses a redefined `psa_key_id_t` is that it was used as the source of the PSA specification document, and this reason no longer applies.

Reply:
Fixed